src/plugins/UrbanDictionary.js

```diff
@@ -0,0 +1,46 @@
+const Plugin = require('../Plugin');
+const request = require('request-promise');
+
+module.exports = class UrbanDictionary extends Plugin {
+    static get plugin() {
+        return {
+            name: "UrbanDictionary",
+            description: "Fetches the Urban Dictionary definition for a supplied word or phrase.",
+            help: ` commands:
+                \`/ud <word>\` returns the definition for a given word or phrase`
+        };
+    }
+
+    get commands() {
+        return {
+            ud: ({args, message}) => {
+                if (args.length === 0) {
+                    this.sendMessage(message.chat.id, "Please provide a search term after /ud");
+                    return;
+                }
+
+                const query = args.join(' ');
+
+                request(`http://api.urbandictionary.com/v0/define?term=${encodeURIComponent(query)}`)
+                    .then(response => {
+                        const data = JSON.parse(response);
+                        /* data.list will be an array, it seems like typically
+                           the most popular defintion is at the zero index though.
+                        */
+                        const def = data.list[0];
+                        const opts = {parse_mode: 'Markdown'};
+                        let msg;
+
+                        if (data.result_type === 'no_results') {
+                            msg = `Sorry, I was unable to find results for: ${args.join(' ')}.`;
+                        } else {
+                            msg =
+                                `*${def.word}*: ${def.definition}\n\n_${def.example}_`;
+                        }
+
+                        this.sendMessage(message.chat.id, msg, opts);
+                    });
```

Comment:
It lacks a .catch statement: if the request fails for whatever reason, displaying a short error message (possibly with an error on the command line, this.log.error/this.log.warn) is more user-friendly than not showing anything.